### ISSUE TITLE
no module named toga_gtk in PR tests

### ISSUE THREAD

**Issue by lielfr** [pybee/toga#303]
So, I've submitted a pull request (#297). BeeKeeper seems to fail, and when looking at the log the reason for failure is this exception: `ModuleNotFoundError: No module named 'toga_gtk'`.
Maybe this is a bug in the comb?

**Comment by DariusMontez** [comment]
@lielfr Thanks for the pull request! When writing tests for  the Core API, it is important to remember that the tests are platform-independent.  When a Core widget, such as toga.Command, is created it calls upon a "factory" to instantiate the underlying platform-dependent implementation. For GTK, this would be the Command class from `src/gtk/command.py`. If you were running on iOS, it would use `src/ios/command.py` as its factory.
However, when writing tests, a "dummy" factory is used. It is basically an object that allows you to call any method on it or access any property, and the dummy pretends that it did what you told it. I'm sure that sounds confusing, maybe take a look at `src/core/tests/font.py`. It's pretty simple but shows the dummy factory in action.

Basically, you'd be looking at something like:
```python
command = toga.Command(... , factory=toga_dummy.factory)
```

I think I made the same mistake on my first PR to Toga, as It's not immediately obvious. The Contributing Guide could probably do a better job of explaining tests.
Also, you might find it helpful to browse through several other tests to get a feel for how they work. Feel free to ask questions if you get stuck, of course.

**Comment by lielfr** [comment]
Well, I didn't understand that from the contributing guide. Thanks a lot!

**Comment by freakboy3742** [comment]
@lielfr Apologies for the confusion! @DariusMontez is absolutely correct about the cause, and the fact that the contribution guide can be improved. I'll submit an update shortly.

**Comment by freakboy3742** [comment]
I've just added PR #304 - let me know what you think!

**Comment by lielfr** [comment]
It's much clearer now. Thanks!
